src/libraries/Microsoft.Extensions.Diagnostics/ref/Microsoft.Extensions.Diagnostics.csproj

```diff
@@ -9,8 +9,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Diagnostics.DiagnosticSource\ref\System.Diagnostics.DiagnosticSource.csproj" />
-    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.DependencyInjection.Abstractions\ref\Microsoft.Extensions.DependencyInjection.Abstractions.csproj" />
+    <InternalsVisibleTo Include="Microsoft.Extensions.Diagnostics.Tests" />
```

Comment:
Also, it looks weird too do it for reference assemblies. Please apply @noahfalk suggestions.

CC @stephentoub

Reply:
Yeah, in general we've tried to avoid InternalsVisibleTo.  We removed it from a bunch of assemblies, and the only ones it still remains in are the few that moved into the repo from other repos and we just haven't prioritized cleaning them out. (InternalsVisibleTo also interferes with some analyzers, that give up when they see it because they're doing whole-assembly validation of things and can't see all consumers of internals.)